### PR TITLE
session,executor: add variables to support set tiflash spill threshold by sql statement

### LINKED ISSUE #42308: Add a way to set TiFlash's spill threshold by sql statement
## Enhancement
After https://github.com/pingcap/tiflash/issues/6528, some TiFlash executor can spill to disk if the memory usage exceeds the threshold, however, the memory usage threshold can only be changed by TiFlash's configure file, which is not easy to use, we'd better add a way to set TiFlash's spill threshold by sql statement.

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #42308

Problem Summary:

### What is changed and how it works?
Add 3 session variable to set TiFlash's spill threshold by sql statement
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -156,6 +156,18 @@ var defaultSysVars = []*SysVar{
 		s.TiFlashMaxThreads = TidbOptInt64(val, DefTiFlashMaxThreads)
 		return nil
 	}},
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBMaxBytesBeforeTiFlashExternalJoin, Type: TypeInt, Value: strconv.Itoa(DefTiFlashMaxBytesBeforeExternalJoin), MinValue: -1, MaxValue: math.MaxInt64, SetSession: func(s *SessionVars, val string) error {
```
Comment (guo-shaoge) [inline sessionctx/variable/sysvar.go:159]: TypeInt means max value is 2G?

Comment (windtalker) [reply sessionctx/variable/sysvar.go:159]: TypeInt should be Int64? I manually tested that set to 10G can work

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**